rust/headless-client/src/dns_control/linux/etc_resolv_conf.rs

```diff
@@ -47,9 +48,13 @@ pub(crate) fn revert() -> Result<()> {
     revert_at_paths(&ResolvPaths::default())
 }
 
-fn configure_at_paths(dns_config: &[IpAddr], paths: &ResolvPaths) -> Result<()> {
-    if dns_config.is_empty() {
-        tracing::warn!("`dns_config` is empty, leaving `/etc/resolv.conf` unchanged");
+fn configure_at_paths(
+    dns_servers: &[IpAddr],
+    _search_domains: &[DomainName],
```

Comment:
This pair of args appears in so many places, maybe we should extract it as a cross-platform `struct DnsConfig`?

Reply:
I am a bit hesitant because that introduces very subtle dependencies. For example, that struct would need to be serialisable so we can pass it across the Android and Swift FFI (we pass JSON strings there).

Yet, for it to be accessible everywhere, that struct would need to be defined in `connlib_shared`, i.e. the dependency root. Dependencies come with a price tag (must be compiled before other code, harder to modify code that ends up having different concerns, etc).

IMO, just bundling together two fields isn't worth paying that price.

Reply:
I am a bit hesitant because that introduces very subtle dependencies. For example, that struct would need to be serialisable so we can pass it across the Android and Swift FFI (we pass JSON strings there).

Yet, for it to be accessible everywhere, that struct would need to be defined in `connlib_shared`, i.e. the dependency root. Dependencies come with a price tag (must be compiled before other code, harder to modify code that ends up having different concerns, etc).

IMO, just bundling together two fields isn't worth paying that price.